src/fred/cli/series.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+"""
+FRED CLI - Series Namespace.
+"""
+import json
+
+import click
+
+from .. import BaseFredAPIError
+from .._util import generate_api_kwargs
+
+__all__ = [
+    "series",
+]
+
+
+@click.group()
+def series():
+    """
+    Series CLI Namespace.
+    """
+    pass
+
+
+@series.command()
+@click.option("--series-id", "-i", required=True, type=click.STRING, help="Series ID.")
+@click.argument("args", nargs=-1)
+@click.pass_context
+def get_series(ctx, series_id: str, args: tuple):
+    """Get series."""
+    try:
+        click.echo(json.dumps(ctx.obj["client"].get_series(series_id, **generate_api_kwargs(args)), indent=4))
+    except (ValueError, BaseFredAPIError) as e:
+        click.echo(click.style(e, fg="red"))
+
+
+@series.command()
+@click.option("--series-id", "-i", required=True, type=click.STRING, help="Series ID.")
+@click.argument("args", nargs=-1)
+@click.pass_context
+def get_series_categories(ctx, series_id: str, args: tuple):
+    """Get series categories."""
+    try:
+        click.echo(
+            json.dumps(ctx.obj["client"].get_series_categories(series_id, **generate_api_kwargs(args)), indent=4)
+        )
+    except (ValueError, BaseFredAPIError) as e:
+        click.echo(click.style(e, fg="red"))
+
+
+@series.command()
+@click.option("--series-id", "-i", required=True, type=click.STRING, help="Series ID.")
+@click.argument("args", nargs=-1)
+@click.pass_context
+def get_series_observations(ctx, series_id: str, args: tuple):
+    """Get series observations."""
+    try:
+        click.echo(
+            json.dumps(ctx.obj["client"].get_series_observations(series_id, **generate_api_kwargs(args)), indent=4)
+        )
+    except (ValueError, BaseFredAPIError) as e:
+        click.echo(click.style(e, fg="red"))
+
+
+@series.command()
+@click.option("--series-id", "-i", required=True, type=click.STRING, help="Series ID.")
+@click.argument("args", nargs=-1)
+@click.pass_context
+def get_series_release(ctx, series_id: str, args: tuple):
+    """Get series release."""
+    try:
+        click.echo(json.dumps(ctx.obj["client"].get_series_release(series_id, **generate_api_kwargs(args)), indent=4))
+    except (ValueError, BaseFredAPIError) as e:
+        click.echo(click.style(e, fg="red"))
+
+
+@series.command()
+@click.option("--search-text", "-t", required=True, type=click.STRING, help="Search text.")
+@click.argument("args", nargs=-1)
+@click.pass_context
+def get_series_search(ctx, search_text: str, args: tuple):
```

Comment:
This is missing a required argument called `search_type; str` which should have a defaulted value of `"full_text"`